lib/publiccloud/instance.pm

```diff
@@ -477,4 +491,179 @@ sub network_speed_test() {
     record_info("curl $rmt_host", $self->run_ssh_command(cmd => "curl -w '$write_out' -o /dev/null -v https://$rmt_host/", proceed_on_failure => 1));
 }
 
+=head2 measure_boottime
+
+    measure_boottime();
+
+Perfomrance measurement of the system Boot time. 
+Mainly used C<systemd-analyze> command for the data extraction.
+Data is then collected in an internal record, ready for storing in a DB.
+Set PUBLIC_CLOUD_PERF_COLLECT true or >0, to activate boottime measurements.
+
+=cut
+
+sub measure_boottime() {
+    my ($self, $instance, $type) = @_;
+    return 0 unless (get_var('PUBLIC_CLOUD_PERF_COLLECT'));
+
+    my $ret = {
+        kernel_release => undef,
+        kernel_version => undef,
+        type => undef,
+        analyze => {},
+        blame => {},
+    };
+
+    record_info("BOOT TIME", 'systemd_analyze');
+    # first deployment analysis
+    my ($systemd_analyze, $systemd_blame) = do_systemd_analyze_time($instance);
+    return 0 unless ($systemd_analyze && $systemd_blame);
+
+    $ret->{analyze}->{$_} = $systemd_analyze->{$_} foreach (keys(%{$systemd_analyze}));
+    $ret->{blame} = $systemd_blame;
+    $ret->{type} = $type;
+    # $ret->{analyze}->{ssh_access} = $startup_time; # placeholder for next implementation
+
+    # Collect kernel version
+    $ret->{kernel_release} = $instance->run_ssh_command(cmd => 'uname -r', proceed_on_failure => 1);
+    $ret->{kernel_version} = $instance->run_ssh_command(cmd => 'uname -v', proceed_on_failure => 1);
+
+    # Do logging to openqa UI
+    $Data::Dumper::Sortkeys = 1;
+    record_info("RESULTS", Dumper($ret));
+
+    my $logs = '/var/log/cloudregister /var/log/cloud-init.log /var/log/cloud-init-output.log /var/log/messages /var/log/NetworkManager';
+    $instance->run_ssh_command(cmd => 'sudo tar -czvf /tmp/sle_cloud.tar.gz ' . $logs, proceed_on_failure => 1);
+    $instance->upload_log('/tmp/sle_cloud.tar.gz', failok => 1);
```

Comment:
Please upload those files one by one. Those archives are slower to review.

Reply:
Sorry, but those files go  into a **tar.gz** that we'll find into the final logs, faster to check then.

That file list was so long already in the original version of [boottime.pm](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/322959a47faef1309f3c4d8395b624eb866f5f2c/tests/publiccloud/boottime.pm#L345) ,before my changes, all together inline into that run_ssh_command. In this new version I only moved that list to the `logs` var, to improve readiness.

To upload those files one-by-one, it means to provide that same list into an additional loop or a map, with no improvement.

Reply:
Yes, that's true. You can do:
```perl
my @logs = ('/var/log/x', '/var/log/y', '/var/log/z');
upload_logs($_) each @logs;
```

Reply:
Ok thanks, or even so :
```
my @logs = qw( /var/log/x   /var/log/y   /var/log/z );
upload_logs($_) each @logs;
```
But, without a containing **tar.gz**, at end of test we have that long number of different files, also unrelated even being collected all in that same point.

Reply:
Yes, please keep the "long number of different files" part :smile:
The openQA has cleanup scripts and as long as the files are properly named it's easier to open them directly than downloading an archive, decompressing it and what not...

Reply:
> Yes, please keep the "long number of different files" part

Well, will do it, but the same then we'll have to _download_ those `*.log` named files for reading.
Openqa UI only open .txt or .json.

Reply:
Yes, please keep the "long number of different files" part

Well, I prefer the _tar_, but I will do it. Anyway the same we'll have to download those *.log named files for reading, because Openqa UI seems to only open .txt or .json files., like [that](https://openqa.suse.de/tests/10611783#downloads)

Reply:
> Yes, please keep the "long number of different files" part

Well, I'd have no problem to handle the _tar_ as a log. Anyway, then, the same we'll have to download those single *.log named files for reading, because Openqa UI seems to only open .txt or .json files., like [that](https://openqa.suse.de/tests/10611783#downloads)

Reply:
done